java-checks/src/main/java/org/sonar/java/checks/CompareStringsBoxedTypesWithEqualsCheck.java

```diff
@@ -19,29 +19,107 @@
+  private static final String QUICK_FIX_MESSAGE = "Replace with boxed comparison";
+  private static final String DOT_EQUALS_AND_OPENING_PARENTHESIS = ".equals(";
+
   @Override
   protected void checkEqualityExpression(BinaryExpressionTree tree) {
     Type leftOpType = tree.leftOperand().symbolType();
     Type rightOpType = tree.rightOperand().symbolType();
     if (!isNullComparison(leftOpType, rightOpType)
       && !isCompareWithBooleanConstant(tree.leftOperand(), tree.rightOperand())
       && (isStringType(leftOpType, rightOpType) || isBoxedType(leftOpType, rightOpType))) {
-      reportIssue(tree.operatorToken());
+      ((InternalJavaIssueBuilder) ((DefaultJavaFileScannerContext) this.context).newIssue())
+        .forRule(this)
+        .onTree(tree.operatorToken())
+        .withMessage(ISSUE_MESSAGE)
+        .withQuickFixes(() -> computeQuickFix(tree))
+        .report();
     }
   }
-  
+
   private static boolean isCompareWithBooleanConstant(ExpressionTree left, ExpressionTree right) {
     return ExpressionsHelper.getConstantValueAsBoolean(left).value() != null ||
       ExpressionsHelper.getConstantValueAsBoolean(right).value() != null;
   }
 
+  private static List<JavaQuickFix> computeQuickFix(BinaryExpressionTree tree) {
+    Optional<JavaQuickFix> conciseQuickFix = computeConciseQuickFix(tree);
+    if (conciseQuickFix.isPresent()) {
+      return Collections.singletonList(conciseQuickFix.get());
+    }
+    return Collections.singletonList(computeDefaultQuickFix(tree));
+  }
+
+  private static Optional<JavaQuickFix> computeConciseQuickFix(BinaryExpressionTree tree) {
+    if (tree.leftOperand().is(Tree.Kind.STRING_LITERAL)) {
+      AnalyzerMessage.TextSpan interOperandSpace = AnalyzerMessage.textSpanBetween(
+        tree.leftOperand().lastToken(), false,
+        tree.rightOperand().firstToken(), false
+      );
+      List<JavaTextEdit> edits = new ArrayList<>();
+      edits.add(JavaTextEdit.insertAfterTree(tree.rightOperand().lastToken(), ")"));
+      edits.add(JavaTextEdit.replaceTextSpan(interOperandSpace, DOT_EQUALS_AND_OPENING_PARENTHESIS));
+      if (tree.is(Tree.Kind.NOT_EQUAL_TO)) {
+        edits.add(JavaTextEdit.insertBeforeTree(tree.leftOperand().firstToken(), "!"));
+      }
+      JavaQuickFix.Builder quickFix = JavaQuickFix.newQuickFix(QUICK_FIX_MESSAGE)
+        .addTextEdit(JavaTextEdit.insertAfterTree(tree.rightOperand().lastToken(), ")"))
+        .addTextEdit(JavaTextEdit.replaceTextSpan(interOperandSpace, DOT_EQUALS_AND_OPENING_PARENTHESIS));
+      if (tree.is(Tree.Kind.NOT_EQUAL_TO)) {
+        quickFix.addTextEdit(JavaTextEdit.insertBeforeTree(tree.leftOperand().firstToken(), "!"));
+      }
+      return Optional.of(quickFix.build());
+    } else if (tree.rightOperand().is(Tree.Kind.STRING_LITERAL)) {
+      String callEqualsOnLiteral = ((LiteralTree) tree.rightOperand()).value() + DOT_EQUALS_AND_OPENING_PARENTHESIS;
```

Comment:
You are calling multiple times `tree.leftOperand()` and `tree.rightOperand()`, you could extract them into a variable before the if/else if.